src/Resque/Redis/RedisWorkerRegistry.php

```diff
@@ -85,9 +90,9 @@ public function deregister(WorkerInterface $worker)
     {
         $id = $worker->getId();
 
-        $worker->halt();
```

Comment:
Workers are deregistered as the result of signals received, i.e from stop(), calling halt() sends a new signal which can be picked up with the signal handler (depending on ticks). This has real world implications when trying to politely shut down workers for the purposes of a graceful restart.

That and its a crossing of boundaries. I don't see why asking for de-registration should result in the worker being sent a signal. The request to halt/stop a worker should instead be resulting in de-registration.